core/src/orderbook/history.rs

```diff
@@ -0,0 +1,98 @@
+//! Module for historic orderbook reading.
+
+use super::{NoopOrderbook, StableXOrderBookReading};
+use crate::models::{AccountState, Order};
+use anyhow::Result;
+use ethcontract::BlockNumber;
+use futures::future::BoxFuture;
+use std::time::SystemTime;
+
+/// Trait for reading historic orderbook data.
+pub trait HistoricOrderbookReading: StableXOrderBookReading {
+    /// Retrieves the open orderbook at the specified block number.
+    ///
+    /// The open orderbook is defined as the auction state for the current batch
+    /// at the specified block number. This implies that, assuming no futher
+    /// changes to the exchange, this will be the auction state passed to the
+    /// solver to be submitted in the next batch.
+    fn get_auction_data_for_block<'a>(
+        &'a self,
+        block_number: BlockNumber,
+    ) -> BoxFuture<'a, Result<(AccountState, Vec<Order>)>>;
+
+    /// Retrieves the open orderbook at the specified timestamp.
+    fn get_auction_data_for_timestamp<'a>(
+        &'a self,
+        timestamp: SystemTime,
+    ) -> BoxFuture<'a, Result<(AccountState, Vec<Order>)>>;
+}
```

Comment:
This could be one method taking an enum BlockNumberOrSystemTime (with a better name). Just an idea.

Reply:
I found the separate functions easier to compose when doing the implementation.